mapcomposer/app/static/externals/gxp/src/script/widgets/ChartBuilder.js

```diff
@@ -0,0 +1,199 @@
+     *  ``Boolean`` Do we allow blank ChartFields? It is safe to say true
+     *  here, but for compatibility reasons with old applications, the default
+     *  is true.
+     */
+    allowBlank: true,
+
+    /** api: config[cls]
+     *  ``String``
+     *  The CSS class to be added to this panel's element (defaults to
+     *  ``"gxp-chartbuilder"``).
+     */
+    cls: "gxp-chartbuilder",
+
+    /** i18n */
+    createChartText: "Generate",
+    histogramChartText: "Histogram",
+    lineChartText: "Line",
+    pieChartText: "Pie",
+    gaugeChartText: "Gauge",
+    chartTypeLabel: "Chart type",
+    xAxisLabel: "X axis",
+    yAxisLabel: "Y axis, Aggregation",
+
+    initComponent: function() {
+
+        this.items = [{
+            xtype: "container",
+            layout: "form",
+            ref: "form",
+            //defaults: {anchor: "100%"},
+            hideLabels: false,
+            labelWidth: 130,
+            items: [
+                this.createChartTypeCombo(),
+                this.createChildChartPanel(),
+                {
+                    xtype: "toolbar",
+                    items: this.createToolBar()
+                }
+            ]
+        }];
+
+        this.addEvents(
+            /**
+             * Event: change
+             * Fires when the filter changes.
+             *
+             * Listener arguments:
+             * builder - {gxp.ChartBuilder} This filter builder.  Call
+             *     ``getFilter`` to get the updated filter.
+             */
+            "change"
+        );
+
+        gxp.ChartBuilder.superclass.initComponent.call(this);
+    },
+
+    /** private: method[createToolBar]
+     */
+    createToolBar: function() {
+        var bar = [{
+            text: this.createChartText,
+            iconCls: "gxp-icon-buildchart",
+            handler: function() {
+                Ext.MessageBox.show({
+                    title: 'Info',
+                    msg: 'This functionality is not yet implemented!',
+                    buttons: Ext.Msg.OK,
+                    animEl: 'elId',
+                    icon: Ext.MessageBox.INFO
+                });
+            },
+            scope: this
+        }];
+        return bar;
+    },
+
+    createChartTypeCombo: function() {
+        var data = [
+            [0,this.histogramChartText],
+            [1,this.lineChartText],
+            [2,this.pieChartText],
+            [3,this.gaugeChartText]
+        ];
+        return {
+            xtype: "combo",
+            fieldLabel: this.chartTypeLabel,
+            store: new Ext.data.SimpleStore({
+                data: data,
+                fields: ["value", "name"]
+            }),
+            value: 0,
+            originalValue: 0,
+            ref: "../chartTypeCombo",
+            displayField: "name",
+            valueField: "value",
+            triggerAction: "all",
+            mode: "local",
+            listeners: {
+                select: function(combo, record) {
```

Comment:
Gauge selected should show additional option max. Other options specific, like highlighted intervals can be set in a second time from chart options.